### ISSUE TITLE
Option (or new default) for disallowing arbitrary senders

### ISSUE THREAD

**Issue by Guite** [zikula-ev/Formicula#63]
Formicula is currently not properly usable with SMTP because it uses different email addresses (of users and defined contacts) as sender address. This is not allowed when using a SMTP server. As a workaround I disabled the `fromaddress` property from the `Mailer` calls in the user api.

**Comment by cmfcmf** [comment]
Formicula works fine for me? Am I missing something?

**Comment by Guite** [comment]
Do you have a SMTP server configured in your Mailer settings? Usually a SMTP server does only allow valid sender addresses for the configured mail account. For example if you added the account for `info@example.org` with valid user and password settings, you may only send mails using `info@example.org` as sender. If now a user tries to send a contact request Formicula tries to send a mail from `user@userdomain.org` to `info@example.org` which is denied by the SMTP server. Also if you use `contact@example.org` in the Formicula contact Formicula tries to send the confirmation mail to the user from `contact@example.org` which is also denied because only `info@example.org` is the valid sender for the configured mail account.

**Comment by cmfcmf** [comment]
> Do you have a SMTP server configured in your Mailer settings? 

Yes.
> Usually a SMTP server does only allow valid sender addresses for the configured mail account. For example if you added the account for info@example.org with valid user and password settings, you may only send mails using info@example.org as sender. If now a user tries to send a contact request Formicula tries to send a mail from user@userdomain.org to info@example.org which is denied by the SMTP server. Also if you use contact@example.org in the Formicula contact Formicula tries to send the confirmation mail to the user from contact@example.org which is also denied because only info@example.org is the valid sender for the configured mail account.

**Comment by Guite** [comment]
A properly configured SMTP server should not allow arbitrary senders though.

**Comment by Guite** [comment]
For the record: the same problem occured again now.

```
The mail system

<info@customer.com>: host mail.customer.com[123.123.123.123] said:
550 456.456.456.456 is not allowed to send mail from user-domain.de (in reply to RCPT
TO command)
```

**Comment by Guite** [comment]
Fixed in d32428f9b57a8612128b61ec07445ef65035ea5d by adding a new modvar named `use_contacts_as_sender`.
